src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -2536,6 +2536,80 @@ BOOST_AUTO_TEST_CASE(GradientSequentialLayerTest)
   BOOST_REQUIRE_LE(CheckGradient(function), 1e-4);
 }
 
+/**
+ * WeightNorm layer numerical gradient test.
+ */
+BOOST_AUTO_TEST_CASE(GradientWeightNormLayerTest)
+{
+  // Linear function gradient instantiation.
+  struct GradientFunction
+  {
+    GradientFunction()
+    {
+      input = arma::randu(10, 1);
+      target = arma::mat("1");
+
+      model = new FFN<NegativeLogLikelihood<>, NguyenWidrowInitialization>();
+      model->Predictors() = input;
+      model->Responses() = target;
+      model->Add<IdentityLayer<> >();
+
```

Comment:
Could you add a linear layer before the weight norm layer so, that we can check the `Backward` function.

Reply:
Done.